src/manager.rs

```diff
@@ -108,26 +109,40 @@ impl Manager {
     pub fn start(config: Arc<Config>) -> impl FnOnce() -> Pin<Box<dyn Future<Output = ()>>> {
         let checker = Arc::new(HttpChecker::new(!config.allow_internal_ips));
 
-        let kafka_overrides = HashMap::from([("compression.type".to_string(), "lz4".to_string())]);
-
-        let kafka_config = KafkaConfig::new_config(
-            config.results_kafka_cluster.to_owned(),
-            Some(kafka_overrides),
-        );
-
-        let producer = Arc::new(KafkaResultsProducer::new(
-            &config.results_kafka_topic,
-            kafka_config,
-        ));
-
-        // XXX: Executor will shutdown once the sender goes out of scope. This will happen once all
-        // referneces of the Sender (executor_sender) are dropped.
-        let (executor_sender, executor_join_handle) = run_executor(
-            config.checker_concurrency,
-            checker,
-            producer,
-            config.region.clone(),
-        );
+        let (executor_sender, (executor_join_handle, results_worker, results_shutdown_sender)) = match &config.producer_mode {
+            ProducerMode::Vector => {
```

Comment:
basically the problem here is that i can't simply set the producer equal to something, as the trait isn't sized? so have to do this weird match right now where we have to duplicate logic. i'm not quite sure what the fix is for that.